src/cryptonote_core/cryptonote_tx_utils.h

```diff
@@ -132,6 +132,13 @@ namespace cryptonote
     , uint32_t nonce
     );
 
+  class Blockchain;
+  bool get_block_longhash(const Blockchain *pb, const block& b, crypto::hash& res, const uint64_t height, const int miners);
+  void get_altblock_longhash(const block& b, crypto::hash& res, const uint64_t main_height, const uint64_t height,
+    const uint64_t seed_height, const crypto::hash& seed_hash);
+  crypto::hash get_block_longhash(const Blockchain *pb, const block& b, const uint64_t height, const int miners);
+  void get_block_longhash_reorg(const uint64_t split_height);
+
```

Comment:
How about another file ? These things have nothing to do with tx utils.

Reply:
Or maybe just move them back to `cryptonote_basic/cryptonote_format_utils.cpp`?

Reply:
No, had dependency cycles which is why I had to pull them out in the first place.

Reply:
I guess I could take a stab at moving this back into cryptonote_basic. @moneromooo-monero you mentioned needing to restructure anyway, to work on MacOSX? Can you explain that again here?

Reply:
cryptonote_core depends on cryptonote_basic and blockchain_db. cryptonote_basic does not depend on blockchain_db. When cryptonote_core and cryptonote_basic were merged, that created a circular dependency, and mac barfed on that. I don't recall the exact symbols causing trouble (not sure I even saw them). stoffu was the one hitting the problem at the time.

FWIW, I think "format" in cryptonote_format_utils is about data format, not output formatting.

Reply:
So currently miner.cpp is in cryptonote_basic; putting this stuff in there will make _basic depend on blockchain_db.

Reply:
Yes, unless you pass a function pointer instead of the blockchainDB pointer.